services/spar/src/Spar/Scim/User.hs

```diff
@@ -65,51 +65,54 @@ import qualified Web.Scim.Schema.User             as Scim.User (schemas)
 -- UserDB instance
 
 instance Scim.UserDB SparTag Spar where
-  -- | List all users, possibly filtered by some predicate.
+  -- | Filters users based on some predicate.  We do not support this endpoint
+  -- without an explicit predicate due to performance reasons.  The predicate
+  -- can currently look for users based on handle or SSO ID. 
+  --
+  -- This endpoint will look for users in Brig, and then synthesize the
+  -- corresponding SCIM record.
+  -- look for SSO users in brig; and then return the corresponding SCIM record.
+  -- getUsers itself will not do any mutations in the SCIM database; as it
+  -- feels weird to me (arian) that a GET causes side-effects like that.
+  -- Instead, 'updateUser' will when updating a user see if the user was
+  -- already in the SCIM Table; and if not insert it on demand, just like
+  -- 'POST'.
```

Comment:
Actually I changed my mind here.

`getUsers` should insert the synthesized record into the scim database; and `updateUsers` should only set `mangedBy = SCIM` on the user.

Reason?  the SCIM record have a `version` and a `modifiedAt` field that need to be correct. The only way to do that is to set them during creation time (I can't come up with a `modifiedAt` out of thing air and then have it the same the next time `updateUser` is called unless I store it).